### PR TITLE
Add docker molecule driver

### PR DESCRIPTION
**Pull request by ziegenberg**
As this image is superseding the ansible toolset image, it should aim at feature parity with its precursor.

This change adds the molecule docker driver, and its necessary dependencies. This also includes the `community.docker` ansible collection, as written in the [molecule-docker README](https://github.com/ansible-community/molecule-docker/blob/main/README.rst).

Partly fixes: #20

### CODE REVIEW

**Comment by ziegenberg** [comment]
hi @ssbarnea!

Could you approve running the ci workflow?

File: _build/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile with python 3.10
```
Comment (ssbarnea) [inline _build/requirements.txt:2]: That should be taken out.

Comment (ziegenberg) [reply _build/requirements.txt:2]: I'll fix that

Comment (ziegenberg) [reply _build/requirements.txt]: done.

**Review by ssbarnea — APPROVED** [review]

**Comment by ziegenberg** [comment]
I think I made a small mistake. `molecule-docker` should be a bit more relaxed on the version and be set to `molecule-docker>=1.1.0`. Same applies to podman.

**Pull request merged**